lib/modules/dosomething/dosomething_campaign/dosomething_campaign.helpers.inc

```diff
@@ -84,190 +86,205 @@
+      // @todo: Deprecate when Message Broker uses url array instead.
+      $campaign->image_cover['src'] = dosomething_image_get_themed_image_url($campaign->image_cover['nid'], $image_src_ratio, $image_src_style);
+      // Set the url array.
+      dosomething_campaign_load_image_url($campaign->image_cover['url'], $image_cover);
     }
-    // Set the src property.
-    // @todo: Deprecate when Message Broker uses url array instead.
-    $campaign->image_cover['src'] = dosomething_image_get_themed_image_url($campaign->image_cover['nid'], $image_src_ratio, $image_src_style);
-    // Set the url array.
-    dosomething_campaign_load_image_url($campaign->image_cover['url'], $image_cover);
-  }
 
-  // Set image_header property.
-  // If there is an override for the campaign header image.
-  if (isset($campaign->variables['alt_image_campaign_cover_nid'])) {
-    // Set the nid to the overridden value.
-    $campaign->image_header['nid'] = (int) $campaign->variables['alt_image_campaign_cover_nid'];
-    // Initalize as FALSE.
-    $campaign->image_header['is_dark_image'] = FALSE;
-    // Load the alt image node.
-    $alt_image = node_load($campaign->image_header['nid']);
-    if (isset($alt_image->field_dark_image[LANGUAGE_NONE][0]['value'])) {
-      $campaign->image_header['is_dark_image'] = (bool) $alt_image->field_dark_image[LANGUAGE_NONE][0]['value'];
+    // Set image_header property.
+    // If there is an override for the campaign header image.
+    if (isset($campaign->variables['alt_image_campaign_cover_nid'])) {
+      // Set the nid to the overridden value.
+      $campaign->image_header['nid'] = (int) $campaign->variables['alt_image_campaign_cover_nid'];
+      // Initalize as FALSE.
+      $campaign->image_header['is_dark_image'] = FALSE;
+      // Load the alt image node.
+      $alt_image = node_load($campaign->image_header['nid']);
+      if (isset($alt_image->field_dark_image[LANGUAGE_NONE][0]['value'])) {
+        $campaign->image_header['is_dark_image'] = (bool) $alt_image->field_dark_image[LANGUAGE_NONE][0]['value'];
+      }
+      // Set the src property.
+      $campaign->image_header['src'] = dosomething_image_get_themed_image_url($campaign->image_header['nid'], $image_src_ratio, $image_src_style);
     }
-    // Set the src property.
-    $campaign->image_header['src'] = dosomething_image_get_themed_image_url($campaign->image_header['nid'], $image_src_ratio, $image_src_style);
-  }
-  // Else if no override:
-  else {
-    // Same as the image_cover.
-    $campaign->image_header = $campaign->image_cover;
-  }
+    // Else if no override:
+    else {
+      // Same as the image_cover.
+      $campaign->image_header = $campaign->image_cover;
+    }
+
+    // Load Creator property.
+    dosomething_campaign_load_creator($campaign, $wrapper);
 
-  // Load Creator property.
-  dosomething_campaign_load_creator($campaign, $wrapper);
+    cache_set('ds_campaign_' . $campaign->nid, $campaign, 'cache_dosomething_campaign');
```

Comment:
Might be good to not cache it forever, but add a time for how long to cache the node for.